src/Tizen.NUI.Components/Controls/NumberPicker.cs

```diff
@@ -0,0 +1,531 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using Tizen.NUI;
+using Tizen.NUI.BaseComponents;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// NumberPickerValueChangedEventArgs is a class to notify changed NumberPicker value argument which will sent to user.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class NumberPickerValueChangedEventArgs : EventArgs
+    {
+        /// <summary>
+        /// NumberPickerValueChangedEventArgs default constructor.
+        /// <param name="value">value of NumberPicker.</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]   
+        public NumberPickerValueChangedEventArgs(int value)
+        {
+            Value = value;
+        }
+
+        /// <summary>
+        /// NumberPickerValueChangedEventArgs default constructor.
+        /// <returns>The current value of NumberPicker.</returns>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]   
+        public int Value { get; }
+        
+    }
+
+    /// <summary>
+    /// NumberPicker is a class which provides a function that allows the user to select 
+    /// a value through a scrolling motion by expressing the specified value as a list.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class NumberPicker : Control
+    {
+        //Tizen 6.5 base components Picker guide visible scroll item is 5.
+        private const int scrollVisibleItems = 5;
+        private const int dummyItemsForLoop = 20; //Dummy item count for loop feature.
+        private int startScrollOffset;
+        private int itemHeight;
+        private int startScrollY;
+        private int startY;
+        private int pageSize;
+        private int currentValue;
+        private int maxValue;
+        private int minValue;
+        private int lastScrollPosion;
+        private bool onAnimation; //Scroller on animation check.
+        private bool displayedValuesUpdate; //User sets displayed value check.
+        private bool needItemUpdate; //min, max or display value updated check.
+        private bool loopEnabled;
+        private ReadOnlyCollection<string> displayedValues;
+        private PickerScroller pickerScroller;
+        private VisualView upLine;
```

Comment:
It seems that both `upLine` and `downLine` don't use visuals. If so `View` is just fine for them instead `VisualView`. (`VisualView` is heavier than `View`)

Reply:
I used to think that the visual view is the lighter than view object.